app/prisma/schema.prisma

```diff
@@ -611,18 +613,70 @@ model UsageLog {
 
     inputTokens  Int
     outputTokens Int
+    inputCost    Decimal @default(0)
+    outputCost   Decimal @default(0)
     cost         Float
     type         UsageType @default(EXTERNAL)
     billable     Boolean   @default(true)
 
     fineTuneId String?   @db.Uuid
     fineTune   FineTune? @relation(fields: [fineTuneId], references: [id], onDelete: SetNull)
+    baseModel  String?
+    projectId  String?  @db.Uuid
+    project    Project? @relation(fields: [projectId], references: [id], onDelete: Cascade)
 
-    projectId String?  @db.Uuid
-    project   Project? @relation(fields: [projectId], references: [id], onDelete: Cascade)
+    invoiceId String?  @db.Uuid
+    invoice  Invoice? @relation(fields: [invoiceId], references: [id], onDelete: SetNull)
 
     createdAt DateTime @default(now())
 
     @@index([fineTuneId, createdAt, type])
     @@index([projectId, createdAt, type])
 }
+
+
+model Invoice {
+    id String @id @default(uuid()) @db.Uuid
+    amount         Decimal @default(0)
+    status         InvoiceStatus @default(PENDING)
+    slug           String
+    billingPeriod  String? // To make invoices more generic. Can we charge clients for something else manually?
+
+    projectId String  @db.Uuid
+    project   Project @relation(fields: [projectId], references: [id], onDelete: Cascade)
+
+    description Json? // Array of objects: { [key: string]: string | number }
+
+    usageLogs UsageLog[]
+
+    createdAt DateTime @default(now())
+    paidAt DateTime?
+
+    paymentId String?    
+
+    remainingCredits Decimal?
```

Comment:
This shouldn't be null I think

Reply:
0 by default should work as well. I think this column is for information purposes only. Should display the "remaining credits" on the invoice if it is not 0?

Reply:
Got it. I thought we would use the creditAdjustment table (as a kind of event sourcing) to calculate the remaining credits.

Reply:
We'll do both. That's where allNewCreditsThisMonth comes into play.

Reply:
It seems like we have 2 sources of truth in this case?

I was thinking of using only the creditsAjustments table, like:
`creditsLeftThisMonth = sum(creditsAjustments) - usageThisMonth.`
Did I understand it wrong?